go/vt/vtgate/evalengine/func.go

```diff
@@ -643,3 +645,48 @@ func (ta *typeAggregation) result() sqltypes.Type {
 	}
 	return sqltypes.VarChar
 }
+
+type builtinCeil struct{}
+
+func (builtinCeil) call(env *ExpressionEnv, args []EvalResult, result *EvalResult) {
+	inarg := &args[0]
+	argtype := inarg.typeof()
+	if inarg.isNull() {
+		result.setNull()
+		return
+	}
+
+	if sqltypes.IsIntegral(argtype) {
+		inarg.makeFloat()
+		result.setInt64(int64(math.Ceil(inarg.float64())))
+	} else if sqltypes.Decimal == argtype {
```

Comment:
This is a small nit, but we don't have any reverse comparisons in the Vitess codebase. As you may be aware of, reverse comparisons have their origin in the C family of languages because `if (argtype = sqltypes.Decimal)` is valid syntax and does something very different than `if (argtype == sqltypes.Decimal)`, so reversing the comparison may prevent a bad bug from being deployed.

However, Go does not allow assignments in `if` expressions, so we always write `if argtype == sqltypes.Decimal {` -- there's no point on doing otherwise except making the code harder to read!